src/lib/comms/include/sol-coap.h

```diff
@@ -78,7 +78,7 @@ extern "C" {
  *
  * Refer to RFC 7252, section 12.2 for more information.
  */
-typedef enum {
+enum sol_coap_option_num_t {
```

Comment:
ugh, -1 as well. if you were really following our patterns you'd remove the typedef (no _t)